SingularityUI/app/views/tail.coffee

```diff
@@ -176,7 +178,8 @@ class TailView extends View
         @$el.addClass 'tailing'
 
     stopTailing: ->
-        return if @isTailing isnt true
+        task = @model.get('taskUpdates')[..].pop()
+        return if @isTailing isnt true and task.taskState in TERMINAL_TASK_STATES
```

Comment:
this logic is confusing to me -- it'd be clearer to modify https://github.com/HubSpot/Singularity/blob/log_never_complete/SingularityUI/app/views/tail.coffee#L36 to call a new method that does one final tail fetch, and then stops the tailing interval